### PR TITLE
Finished Landlord Messaging Feature

### PR DESCRIPTION
**Pull request by parsa-tehranipoor**
### Summary 

This pull request is split into two; the first half consists of the work I completed last semester, and the second half consists of the work I have added since then.

The first part of this PR includes all of the code for this feature's frontend on both mobile and computer devices. This PR also consists of the backend route needed for sending an email out to the landlord using the CUAPTs team email.

- [x] Backend route to send an email to the landlord
- [x] First frontend display on laptop with all features for email creation
- [x] Second frontend display on laptop with all features for confirming email send
- [x] Mobile integration of frontend

The second part of this PR involves fixing multiple backend bugs as well as polishing up the frontend. A test plan for the entire feature, ensuring emails are sent properly, is also included.

- [x] UX Bugs fixed
- [x] Backend bugs fixed
- [x] Calendar pop-up bug fixed
- [x] Non-existence of landlord email corner case
- [x] Testing plan for email being properly sent out

### Test Plan 

The backend route was successfully tested using my friend's email in place of the landlord's email to make sure I didn't accidentally send any test emails to the landlord. My friend successfully received an email from the CUAPTS email and was able to respond back to my personal email by replying to the previous email.

Image 10-1-25 at 12 35 AM
Image 10-1-25 at 12 36 AM

For testing out the frontend, here are screenshots of what the display looks like for different cases

Screenshot 2025-05-27 at 12 51 36 AM
Screenshot 2025-10-01 at 2 14 32 PM
Screenshot 2025-10-01 at 2 15 30 PM
Screenshot 2025-10-01 at 2 16 22 PM
Screenshot 2025-05-27 at 12 55 55 AM
Screenshot 2025-05-27 at 12 56 00 AM
Screenshot 2025-05-27 at 12 56 25 AM

### CODE REVIEW

File: frontend/src/components/Apartment/LandlordMessagingModal.tsx [hunk 1]
```diff
@@ -0,0 +1,982 @@
...
+  const [phoneError, setPhoneError] = useState(false);
+  const [dateError, setDateError] = useState(false);
+  const [subjectError, setSubjectError] = useState(false);
+  const [messageError, setMessageError] = useState(false);
+
+  const handleInquiryType = (inquiryType: string) => {
+    if (inquiryType == 'availability') {
+      setSubject('Inquiry for ' + aptName + ' Rental Availabilities');
+      setMessage(
+        'Dear ' +
```
Comment (laurenp-2) [inline frontend/src/components/Apartment/LandlordMessagingModal.tsx:410]: You could consider using template literals for better readability here.

File: frontend/src/components/Apartment/ConfirmLandlordMessagingModal.tsx [hunk 2]
```diff
@@ -0,0 +1,331 @@
...
+}: ConfirmLandlordMessagingModalProps) => {
+  const classes = useStyles({ isMobile });
+  const [textSubject, setSubject] = useState(subject);
+  const [message, setMessage] = useState(body);
+
+  const handleSubmit = async () => {
+    try {
+      // Check if email is null
+      if (email === null) {
+        console.error('No contact information found');
```
Comment (laurenp-2) [inline frontend/src/components/Apartment/ConfirmLandlordMessagingModal.tsx:193]: Adding an error toast here would let users know that an error occurred and that their email didn't go through.

File: frontend/src/components/Apartment/LandlordMessagingModal.tsx [hunk 3]
```diff
@@ -0,0 +1,982 @@
...
+                display: 'flex',
+                gap: '24px',
+                flexDirection: isMobile ? 'column' : 'row',
+                marginBottom: '24px',
+              }}
+            >
+              <Box style={{ flex: isMobile ? 1 : '0 0 206px' }}>
+                <Box style={{ display: 'flex' }}>
+                  <Typography className={classes.secondTitleBox}>Name</Typography>
+                  {subjectError && (
```
Comment (laurenp-2) [inline frontend/src/components/Apartment/LandlordMessagingModal.tsx]: You're checking subjectError for the name field validation here, but this should be nameError. The same issue occurs on lines 562 (should use phoneError) and 598 (should use dateError).

**Review by laurenp-2 — COMMENTED** [review]
This PR is very comprehensive and well organized! The code is clear and readable, and the modal flow works smoothly. I've left comments on a few small issues—mainly some form validation bugs (wrong error states being checked) and also some error handling—that can be easily fixed. Overall it looks good!

File: frontend/src/components/Apartment/ConfirmLandlordMessagingModal.tsx (hunk 2 above)
Comment (CasperL1218) [reply frontend/src/components/Apartment/ConfirmLandlordMessagingModal.tsx:193]: Great catch! We will be working to create an alternative modal that allows users to direct to the landlord sight. The toast message could also be a good approach.

**Review by CasperL1218 — CHANGES_REQUESTED** [review]
Great job with implementing the feature, the design was spot on too. As pointed out, it will be great to add some form of notification to the user to indicate the absence of the landlord email. Some approach proposed include:

- A toast message
- An alternative modal (e.g. linking to the apartment/landlord website)

For the testings, please also include some notes about each of the image/image groups. This will help organize tests and clearly indicate requirement achievements.

Overall good job!

**Review by CasperL1218 — APPROVED** [review]
- Nice fix with that calendar popup, you resolved the problem elegantly and it turns out to be pretty neat with the centered modal style. Even though it is not the same as the design, it could be beneficial, especially under the mobile screen layout. It is nice that we do not need to switch out the UI choice or conduct upgrades, so this really shows how deep understanding or good research of the UI kit you are using is important.

- Overall welldone with completing the feature, excited to see it in production after we resolve the node version issue. Good job!
